// plugin-welcome/src/api/models/ControllersCourse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersCourse
 */
export interface ControllersCourse {
    /**
     * 
     * @type {string}
     * @memberof ControllersCourse
     */
    coursename?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersCourse
     */
    degree?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersCourse
     */
    faculty?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersCourse
     */
    institution?: number;
}

export function ControllersCourseFromJSON(json: any): ControllersCourse {
    return ControllersCourseFromJSONTyped(json, false);
}

export function ControllersCourseFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersCourse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'coursename': !exists(json, 'coursename') ? undefined : json['coursename'],
        'degree': !exists(json, 'degree') ? undefined : json['degree'],
        'faculty': !exists(json, 'faculty') ? undefined : json['faculty'],
        'institution': !exists(json, 'institution') ? undefined : json['institution'],
    };
}

export function ControllersCourseToJSON(value?: ControllersCourse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'coursename': value.coursename,
        'degree': value.degree,
        'faculty': value.faculty,
        'institution': value.institution,
    };
}
