// plugin-welcome/src/api/models/EntActivity.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntActivityEdges,
    EntActivityEdgesFromJSON,
    EntActivityEdgesFromJSONTyped,
    EntActivityEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntActivity
 */
export interface EntActivity {
    /**
     * ACTIVITYNAME holds the value of the "ACTIVITYNAME" field.
     * @type {string}
     * @memberof EntActivity
     */
    aCTIVITYNAME?: string;
    /**
     * ADDED holds the value of the "ADDED" field.
     * @type {string}
     * @memberof EntActivity
     */
    aDDED?: string;
    /**
     * HOURS holds the value of the "HOURS" field.
     * @type {number}
     * @memberof EntActivity
     */
    hOURS?: number;
    /**
     * 
     * @type {EntActivityEdges}
     * @memberof EntActivity
     */
    edges?: EntActivityEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntActivity
     */
    id?: number;
}

export function EntActivityFromJSON(json: any): EntActivity {
    return EntActivityFromJSONTyped(json, false);
}

export function EntActivityFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntActivity {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'aCTIVITYNAME': !exists(json, 'ACTIVITYNAME') ? undefined : json['ACTIVITYNAME'],
        'aDDED': !exists(json, 'ADDED') ? undefined : json['ADDED'],
        'hOURS': !exists(json, 'HOURS') ? undefined : json['HOURS'],
        'edges': !exists(json, 'edges') ? undefined : EntActivityEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntActivityToJSON(value?: EntActivity | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'ACTIVITYNAME': value.aCTIVITYNAME,
        'ADDED': value.aDDED,
        'HOURS': value.hOURS,
        'edges': EntActivityEdgesToJSON(value.edges),
        'id': value.id,
    };
}
