// plugin-welcome/src/api/models/EntDegree.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntDegreeEdges,
    EntDegreeEdgesFromJSON,
    EntDegreeEdgesFromJSONTyped,
    EntDegreeEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntDegree
 */
export interface EntDegree {
    /**
     * Degree holds the value of the "degree" field.
     * @type {string}
     * @memberof EntDegree
     */
    degree?: string;
    /**
     * 
     * @type {EntDegreeEdges}
     * @memberof EntDegree
     */
    edges?: EntDegreeEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntDegree
     */
    id?: number;
}

export function EntDegreeFromJSON(json: any): EntDegree {
    return EntDegreeFromJSONTyped(json, false);
}

export function EntDegreeFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntDegree {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'degree': !exists(json, 'degree') ? undefined : json['degree'],
        'edges': !exists(json, 'edges') ? undefined : EntDegreeEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntDegreeToJSON(value?: EntDegree | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'degree': value.degree,
        'edges': EntDegreeEdgesToJSON(value.edges),
        'id': value.id,
    };
}
