// plugin-welcome/src/api/models/EntEmp.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface EntEmp
 */
export interface EntEmp {
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntEmp
     */
    id?: number;
    /**
     * Pass holds the value of the "pass" field.
     * @type {string}
     * @memberof EntEmp
     */
    pass?: string;
    /**
     * Role holds the value of the "role" field.
     * @type {string}
     * @memberof EntEmp
     */
    role?: string;
    /**
     * User holds the value of the "user" field.
     * @type {string}
     * @memberof EntEmp
     */
    user?: string;
}

export function EntEmpFromJSON(json: any): EntEmp {
    return EntEmpFromJSONTyped(json, false);
}

export function EntEmpFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntEmp {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'id': !exists(json, 'id') ? undefined : json['id'],
        'pass': !exists(json, 'pass') ? undefined : json['pass'],
        'role': !exists(json, 'role') ? undefined : json['role'],
        'user': !exists(json, 'user') ? undefined : json['user'],
    };
}

export function EntEmpToJSON(value?: EntEmp | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'id': value.id,
        'pass': value.pass,
        'role': value.role,
        'user': value.user,
    };
}
