// plugin-welcome/src/api/models/EntFacultyEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntCourse,
    EntCourseFromJSON,
    EntCourseFromJSONTyped,
    EntCourseToJSON,
    EntProfessor,
    EntProfessorFromJSON,
    EntProfessorFromJSONTyped,
    EntProfessorToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntFacultyEdges
 */
export interface EntFacultyEdges {
    /**
     * FacuCour holds the value of the facu_cour edge.
     * @type {Array<EntCourse>}
     * @memberof EntFacultyEdges
     */
    facuCour?: Array<EntCourse>;
    /**
     * FacuProf holds the value of the facu_prof edge.
     * @type {Array<EntProfessor>}
     * @memberof EntFacultyEdges
     */
    facuProf?: Array<EntProfessor>;
}

export function EntFacultyEdgesFromJSON(json: any): EntFacultyEdges {
    return EntFacultyEdgesFromJSONTyped(json, false);
}

export function EntFacultyEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntFacultyEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'facuCour': !exists(json, 'facuCour') ? undefined : ((json['facuCour'] as Array<any>).map(EntCourseFromJSON)),
        'facuProf': !exists(json, 'facuProf') ? undefined : ((json['facuProf'] as Array<any>).map(EntProfessorFromJSON)),
    };
}

export function EntFacultyEdgesToJSON(value?: EntFacultyEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'facuCour': value.facuCour === undefined ? undefined : ((value.facuCour as Array<any>).map(EntCourseToJSON)),
        'facuProf': value.facuProf === undefined ? undefined : ((value.facuProf as Array<any>).map(EntProfessorToJSON)),
    };
}
