// plugin-welcome/src/api/models/EntGenderEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntStudent,
    EntStudentFromJSON,
    EntStudentFromJSONTyped,
    EntStudentToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntGenderEdges
 */
export interface EntGenderEdges {
    /**
     * GendStud holds the value of the gend_stud edge.
     * @type {Array<EntStudent>}
     * @memberof EntGenderEdges
     */
    gendStud?: Array<EntStudent>;
}

export function EntGenderEdgesFromJSON(json: any): EntGenderEdges {
    return EntGenderEdgesFromJSONTyped(json, false);
}

export function EntGenderEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntGenderEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'gendStud': !exists(json, 'gendStud') ? undefined : ((json['gendStud'] as Array<any>).map(EntStudentFromJSON)),
    };
}

export function EntGenderEdgesToJSON(value?: EntGenderEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'gendStud': value.gendStud === undefined ? undefined : ((value.gendStud as Array<any>).map(EntStudentToJSON)),
    };
}
