// plugin-welcome/src/api/models/EntProvince.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntProvinceEdges,
    EntProvinceEdgesFromJSON,
    EntProvinceEdgesFromJSONTyped,
    EntProvinceEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntProvince
 */
export interface EntProvince {
    /**
     * District holds the value of the "district" field.
     * @type {string}
     * @memberof EntProvince
     */
    district?: string;
    /**
     * 
     * @type {EntProvinceEdges}
     * @memberof EntProvince
     */
    edges?: EntProvinceEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntProvince
     */
    id?: number;
    /**
     * Postal holds the value of the "postal" field.
     * @type {number}
     * @memberof EntProvince
     */
    postal?: number;
    /**
     * Province holds the value of the "province" field.
     * @type {string}
     * @memberof EntProvince
     */
    province?: string;
    /**
     * Subdistrict holds the value of the "subdistrict" field.
     * @type {string}
     * @memberof EntProvince
     */
    subdistrict?: string;
}

export function EntProvinceFromJSON(json: any): EntProvince {
    return EntProvinceFromJSONTyped(json, false);
}

export function EntProvinceFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntProvince {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'district': !exists(json, 'district') ? undefined : json['district'],
        'edges': !exists(json, 'edges') ? undefined : EntProvinceEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'postal': !exists(json, 'postal') ? undefined : json['postal'],
        'province': !exists(json, 'province') ? undefined : json['province'],
        'subdistrict': !exists(json, 'subdistrict') ? undefined : json['subdistrict'],
    };
}

export function EntProvinceToJSON(value?: EntProvince | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'district': value.district,
        'edges': EntProvinceEdgesToJSON(value.edges),
        'id': value.id,
        'postal': value.postal,
        'province': value.province,
        'subdistrict': value.subdistrict,
    };
}
