// plugin-welcome/src/api/models/EntPrefixEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntProfessor,
    EntProfessorFromJSON,
    EntProfessorFromJSONTyped,
    EntProfessorToJSON,
    EntStudent,
    EntStudentFromJSON,
    EntStudentFromJSONTyped,
    EntStudentToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPrefixEdges
 */
export interface EntPrefixEdges {
    /**
     * PreProf holds the value of the pre_prof edge.
     * @type {Array<EntProfessor>}
     * @memberof EntPrefixEdges
     */
    preProf?: Array<EntProfessor>;
    /**
     * PrefStud holds the value of the pref_stud edge.
     * @type {Array<EntStudent>}
     * @memberof EntPrefixEdges
     */
    prefStud?: Array<EntStudent>;
}

export function EntPrefixEdgesFromJSON(json: any): EntPrefixEdges {
    return EntPrefixEdgesFromJSONTyped(json, false);
}

export function EntPrefixEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPrefixEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'preProf': !exists(json, 'preProf') ? undefined : ((json['preProf'] as Array<any>).map(EntProfessorFromJSON)),
        'prefStud': !exists(json, 'prefStud') ? undefined : ((json['prefStud'] as Array<any>).map(EntStudentFromJSON)),
    };
}

export function EntPrefixEdgesToJSON(value?: EntPrefixEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'preProf': value.preProf === undefined ? undefined : ((value.preProf as Array<any>).map(EntProfessorToJSON)),
        'prefStud': value.prefStud === undefined ? undefined : ((value.prefStud as Array<any>).map(EntStudentToJSON)),
    };
}
