// plugin-welcome/src/api/models/EntRegionEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntProvince,
    EntProvinceFromJSON,
    EntProvinceFromJSONTyped,
    EntProvinceToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntRegionEdges
 */
export interface EntRegionEdges {
    /**
     * RegiProv holds the value of the regi_prov edge.
     * @type {Array<EntProvince>}
     * @memberof EntRegionEdges
     */
    regiProv?: Array<EntProvince>;
}

export function EntRegionEdgesFromJSON(json: any): EntRegionEdges {
    return EntRegionEdgesFromJSONTyped(json, false);
}

export function EntRegionEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntRegionEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'regiProv': !exists(json, 'regiProv') ? undefined : ((json['regiProv'] as Array<any>).map(EntProvinceFromJSON)),
    };
}

export function EntRegionEdgesToJSON(value?: EntRegionEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'regiProv': value.regiProv === undefined ? undefined : ((value.regiProv as Array<any>).map(EntProvinceToJSON)),
    };
}
