// plugin-welcome/src/api/models/EntResults.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntResultsEdges,
    EntResultsEdgesFromJSON,
    EntResultsEdgesFromJSONTyped,
    EntResultsEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntResults
 */
export interface EntResults {
    /**
     * 
     * @type {EntResultsEdges}
     * @memberof EntResults
     */
    edges?: EntResultsEdges;
    /**
     * Grade holds the value of the "grade" field.
     * @type {number}
     * @memberof EntResults
     */
    grade?: number;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntResults
     */
    id?: number;
}

export function EntResultsFromJSON(json: any): EntResults {
    return EntResultsFromJSONTyped(json, false);
}

export function EntResultsFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntResults {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntResultsEdgesFromJSON(json['edges']),
        'grade': !exists(json, 'grade') ? undefined : json['grade'],
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntResultsToJSON(value?: EntResults | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntResultsEdgesToJSON(value.edges),
        'grade': value.grade,
        'id': value.id,
    };
}
