// plugin-welcome/src/api/models/EntTermEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntActivity,
    EntActivityFromJSON,
    EntActivityFromJSONTyped,
    EntActivityToJSON,
    EntResults,
    EntResultsFromJSON,
    EntResultsFromJSONTyped,
    EntResultsToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntTermEdges
 */
export interface EntTermEdges {
    /**
     * TermActi holds the value of the term_acti edge.
     * @type {Array<EntActivity>}
     * @memberof EntTermEdges
     */
    termActi?: Array<EntActivity>;
    /**
     * TermResu holds the value of the term_resu edge.
     * @type {Array<EntResults>}
     * @memberof EntTermEdges
     */
    termResu?: Array<EntResults>;
}

export function EntTermEdgesFromJSON(json: any): EntTermEdges {
    return EntTermEdgesFromJSONTyped(json, false);
}

export function EntTermEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntTermEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'termActi': !exists(json, 'termActi') ? undefined : ((json['termActi'] as Array<any>).map(EntActivityFromJSON)),
        'termResu': !exists(json, 'termResu') ? undefined : ((json['termResu'] as Array<any>).map(EntResultsFromJSON)),
    };
}

export function EntTermEdgesToJSON(value?: EntTermEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'termActi': value.termActi === undefined ? undefined : ((value.termActi as Array<any>).map(EntActivityToJSON)),
        'termResu': value.termResu === undefined ? undefined : ((value.termResu as Array<any>).map(EntResultsToJSON)),
    };
}
