// @backstage/cli/dist/cjs/index-e080fd1a.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var program = _interopDefault(require('commander'));
var chalk2 = _interopDefault(require('chalk'));
var fs = _interopDefault(require('fs-extra'));
var cliCommon = require('@backstage/cli-common');

class CustomError extends Error {
  get name() {
    return this.constructor.name;
  }
}
class ExitCodeError extends CustomError {
  constructor(code, command) {
    if (command) {
      super(`Command '${command}' exited with code ${code}`);
    } else {
      super(`Child exited with code ${code}`);
    }
    this.code = code;
  }
}
function exitWithError(error) {
  if (error instanceof ExitCodeError) {
    process.stderr.write(`
${chalk2.red(error.message)}

`);
    process.exit(error.code);
  } else {
    process.stderr.write(`
${chalk2.red(`${error}`)}

`);
    process.exit(1);
  }
}

const paths = cliCommon.findPaths(__dirname);

function findVersion() {
  const pkgContent = fs.readFileSync(paths.resolveOwn("package.json"), "utf8");
  return JSON.parse(pkgContent).version;
}
const version = findVersion();
const isDev = fs.pathExistsSync(paths.resolveOwn("src"));

const main = (argv) => {
  program.name("backstage-cli").version(version);
  program.command("app:build").description("Build an app for a production release").option("--stats", "Write bundle stats to output directory").action(lazyAction(() => Promise.resolve().then(function () { return require('./build-df4a2416.js'); }), "default"));
  program.command("app:serve").description("Serve an app for local development").option("--check", "Enable type checking and linting").action(lazyAction(() => Promise.resolve().then(function () { return require('./serve-aeda6310.js'); }), "default"));
  program.command("backend:build").description("Build a backend plugin").action(lazyAction(() => Promise.resolve().then(function () { return require('./build-f08eb021.js'); }), "default"));
  program.command("backend:build-image <image-tag>").description("Builds a docker image from the package, with all local deps included").action(lazyAction(() => Promise.resolve().then(function () { return require('./buildImage-f7a1b4a6.js'); }), "default"));
  program.command("backend:dev").description("Start local development server with HMR for the backend").option("--check", "Enable type checking and linting").action(lazyAction(() => Promise.resolve().then(function () { return require('./dev-867a0eaa.js'); }), "default"));
  program.command("app:diff").option("--check", "Fail if changes are required").option("--yes", "Apply all changes").description("Diff an existing app with the creation template").action(lazyAction(() => Promise.resolve().then(function () { return require('./diff-54b547cf.js'); }), "default"));
  program.command("create-plugin").description("Creates a new plugin in the current repository").action(lazyAction(() => Promise.resolve().then(function () { return require('./createPlugin-c7bef644.js'); }), "default"));
  program.command("remove-plugin").description("Removes plugin in the current repository").action(lazyAction(() => Promise.resolve().then(function () { return require('./removePlugin-71b65894.js'); }), "default"));
  program.command("plugin:build").description("Build a plugin").action(lazyAction(() => Promise.resolve().then(function () { return require('./build-df567f14.js'); }), "default"));
  program.command("plugin:serve").description("Serves the dev/ folder of a plugin").option("--check", "Enable type checking and linting").action(lazyAction(() => Promise.resolve().then(function () { return require('./serve-021645bb.js'); }), "default"));
  program.command("plugin:export").description("Exports the dev/ folder of a plugin").option("--stats", "Write bundle stats to output directory").action(lazyAction(() => Promise.resolve().then(function () { return require('./export-bb964b80.js'); }), "default"));
  program.command("plugin:diff").option("--check", "Fail if changes are required").option("--yes", "Apply all changes").description("Diff an existing plugin with the creation template").action(lazyAction(() => Promise.resolve().then(function () { return require('./diff-d9805adb.js'); }), "default"));
  program.command("build").description("Build a package for publishing").option("--outputs <formats>", "List of formats to output [types,cjs,esm]").action(lazyAction(() => Promise.resolve().then(function () { return require('./build-204fad71.js'); }), "default"));
  program.command("lint").option("--fix", "Attempt to automatically fix violations").description("Lint a package").action(lazyAction(() => Promise.resolve().then(function () { return require('./lint-9ac8b0a6.js'); }), "default"));
  program.command("test").allowUnknownOption(true).helpOption(", --backstage-cli-help").description("Run tests, forwarding args to Jest, defaulting to watch mode").action(lazyAction(() => Promise.resolve().then(function () { return require('./testCommand-80acab7a.js'); }), "default"));
  program.command("config:print").option("--with-secrets", "Include secrets in the printed configuration").option("--env <env>", "The environment to print configuration for [NODE_ENV or development]").option("--format <format>", "Format to print the configuration in, either json or yaml [yaml]").description("Print the app configuration for the current package").action(lazyAction(() => Promise.resolve().then(function () { return require('./print-036182a3.js'); }), "default"));
  program.command("prepack").description("Prepares a package for packaging before publishing").action(lazyAction(() => Promise.resolve().then(function () { return require('./pack-8264ba75.js'); }), "pre"));
  program.command("postpack").description("Restores the changes made by the prepack command").action(lazyAction(() => Promise.resolve().then(function () { return require('./pack-8264ba75.js'); }), "post"));
  program.command("clean").description("Delete cache directories").action(lazyAction(() => Promise.resolve().then(function () { return require('./clean-8888c69d.js'); }), "default"));
  program.command("build-workspace <workspace-dir> ...<packages>").description("Builds a temporary dist workspace from the provided packages").action(lazyAction(() => Promise.resolve().then(function () { return require('./buildWorkspace-0ce2838d.js'); }), "default"));
  program.on("command:*", () => {
    console.log();
    console.log(chalk2.red(`Invalid command: ${chalk2.cyan(program.args.join(" "))}`));
    console.log(chalk2.red("See --help for a list of available commands."));
    console.log();
    process.exit(1);
  });
  if (!process.argv.slice(2).length) {
    program.outputHelp(chalk2.yellow);
  }
  program.parse(argv);
};
function lazyAction(actionRequireFunc, exportName) {
  return async (...args) => {
    try {
      const module = await actionRequireFunc();
      const actionFunc = module[exportName];
      await actionFunc(...args);
      process.exit(0);
    } catch (error) {
      exitWithError(error);
    }
  };
}
process.on("unhandledRejection", (rejection) => {
  if (rejection instanceof Error) {
    exitWithError(rejection);
  } else {
    exitWithError(new Error(`Unknown rejection: '${rejection}'`));
  }
});
main(process.argv);

exports.ExitCodeError = ExitCodeError;
exports.paths = paths;
exports.version = version;
